### ISSUE TITLE
Upgrade AWS SDK used by the repository-s3 plugin to allow use of IMDSv2

### ISSUE THREAD

**Issue by tophercullen** [opensearch-project/OpenSearch#3690]
**Is your feature request related to a problem? Please describe.**
Currently, repository-s3 plugin does not support IMDSv2 due to (I assume) using an older bundled AWS SDK. This means the plugin does not function while using IAM roles and IMDSv2 is enforced.

**Describe the solution you'd like**
Upgrade the AWS SDK used by the plugin.

**Describe alternatives you've considered**
Use AWS static keys/secrets (poor practice). 

**Additional context**
Currently, I am using OS 1.3.2. However, even the 2.x versions I've checked, are using the same, older SDK in the plugin.

**Comment by saratvemulapalli** [comment]
Thanks for opening this up. I've sent out a PR.

**Comment by tophercullen** [comment]
I know its been over a year, but the behavior appears to be unchanged even in 2.7.0.  Perhaps the AWS SDK needs to be update to v2 instead of just a newer version of v1? See #3040

**Comment by dblock** [comment]
The good news is that we upgraded to AWS SDK v2 in https://github.com/opensearch-project/OpenSearch/pull/7372, and https://github.com/opensearch-project/OpenSearch/pull/7372/ which ships with 2.9.0. Want to try the latest release candidate? See https://github.com/opensearch-project/opensearch-build/issues/3616 for latest links.

**Comment by tophercullen** [comment]
Realized I forgot to get back on this. It still doesn't work. We upgraded to OS 2.9.0 late last year and enforcing IMDSv2, the s3 plugin started failing on the next token refresh. We had to re-enable IMDSv1 to get it working again.

**Comment by dblock** [comment]
@tophercullen care to open a new issue for support of IMDSv2 and collect some links/repro steps/whatever else you can provide that shows that it doesn't work?

**Comment by tophercullen** [comment]
Understood. We recently upgrade to 2.12. I'll re-test against that and open an new issue is still present.

**Comment by tophercullen** [comment]
@dblock I've retested this with 2.12 and it works.


As I went about testing it this time around, I believe I found out there was some confusion and/or misunderstanding on my part when testing this previously on 2.9.0. Its entirely possible 2.9.0 works, but at the time I had the EC2 metadata hop limit set to 1. Given I run opensearch from a container, this would cause it to fail when using IMDSv2 and the SDK to fall back to IMDSv1.
